src/circuit.rs

```diff
@@ -1,41 +1,85 @@
+    fn empty() -> Self;
+}
+
+/// A chip's loaded configuration (i.e. lookups, pre-computed fixed constants).
+pub trait Loaded: fmt::Debug + Clone {
+    /// An empty loaded configuration
+    fn empty() -> Self;
+}
+
+impl Loaded for () {
+    fn empty() -> Self {}
+}
+
 /// A chip implements a set of instructions that can be used by gadgets.
 ///
-/// The chip itself should not store any state; instead, state that is required at circuit
-/// synthesis time should be stored in [`Chip::Config`], which can then be fetched via
-/// [`Layouter::config`].
-pub trait Chip: Sized {
+/// The chip stores state that is required at circuit synthesis time in
+/// [`Chip::Config`], which can be fetched via [`Chip::config`].
+///
+/// The chip also loads any fixed configuration needed at synthesis time
+/// using [`Chip::load`], and stores it in [`Chip::Loaded`]. This can be
+/// accessed via [`Chip::loaded`].
+pub trait Chip<F: FieldExt>: Sized {
     /// A type that holds the configuration for this chip, and any other state it may need
     /// during circuit synthesis, that can be derived during [`Circuit::configure`].
     ///
     /// [`Circuit::configure`]: crate::plonk::Circuit::configure
-    type Config: fmt::Debug;
+    type Config: Config;
 
     /// A type that holds any general chip state that needs to be loaded at the start of
     /// [`Circuit::synthesize`]. This might simply be `()` for some chips.
     ///
     /// [`Circuit::synthesize`]: crate::plonk::Circuit::synthesize
-    type Loaded: fmt::Debug;
+    type Loaded: Loaded;
+
+    /// A new chip that is not yet configured or loaded.
+    fn new() -> Self;
 
-    /// The field that the chip is defined over.
+    /// A chip constructed from a given config and loaded state.
+    fn construct(config: Self::Config, loaded: Self::Loaded) -> Self;
+
+    /// The chip is responsible for mutating the constraint system to set up
+    /// the columns, gates, lookups and permutations it needs.
     ///
-    /// This provides a type that the chip's configuration can reference if necessary.
-    type Field: FieldExt;
+    /// We allow for the chip to use pre-existing columns, selectors, and
+    /// permutations in its configuration. Chips that are lower in a hierarchy
+    /// depend on other chips to pass down these objects.
+    fn configure(
+        &mut self,
+        meta: &mut ConstraintSystem<F>,
+        selectors: BTreeMap<&str, Selector>,
+        columns: BTreeMap<&str, Column<Any>>,
+        perms: BTreeMap<&str, Permutation>,
+    ) -> Self::Config;
 
-    /// Load any fixed configuration for this chip into the circuit.
+    /// The chip holds its own configuration.
+    fn config(&self) -> &Self::Config;
+
+    /// Load any fixed configuration for this chip into the circuit, i.e.
+    /// assigns cells in fixed columns.
+    ///
+    /// `self.loaded()` will panic if called inside this function.
+    fn load(&mut self, layouter: &mut impl Layouter<F>) -> Result<Self::Loaded, Error>;
```

Comment:
Now that this is a method on `Chip`, if it is going to take `&mut self` then the type signature should be:
```suggestion
    fn load(&mut self, layouter: &mut impl Layouter<F>) -> Result<(), Error>;
```